app/models/blog.rb

```diff
@@ -7,10 +7,14 @@ class Blog < ApplicationRecord
 
   validates :title, :content, presence: true
 
+  scope :visible_to_user, ->(current_user) { current_user ? owned_by(current_user).or(published) : published }
+
+  scope :owned_by, ->(current_user) { current_user ? where(user_id: current_user.id) : none }
```

Comment:
```suggestion
  scope :owned_by, ->(current_user) { where(user: current_user) }
```

実は上のような書き方でも得られる結果は同じになります。
ログに出力されたSQLを見て、なぜこれでも大丈夫なのか、文章で説明してみてほしいです〜。

Reply:
- `current_user`が渡されているかをチェックする必要が無い理由は、`owned_by`スコープは`visible_to_user`スコープ内でのみ呼び出されていて、そのスコープ内で`current_user`が存在することが保証されているから。
- idではなく直接ユーザを指定しても問題ない理由は、BlogモデルはUserモデルに子の関係で紐付けられており、直接Userモデルとカレントユーザーを比較できて発行されるクエリは殆ど変わらないから。

の上記2つでしょうか、、？

Reply:
あー、`visible_to_user` も条件分岐してたんですね。見落としてました。
実はそっちも条件分岐不要です。

```ruby
scope :visible_to_user, ->(current_user) { owned_by(current_user).or(published) }
```

条件分岐しなくても発行されるSQLを見れば、条件分岐が不要になる理由がわかります。
ということで、SQLを見ながら「なぜ条件分岐がなくても期待どおりに動作するのか」を答えてみてほしいです〜。